### PR TITLE
Fixed tests for `email_notifier.py`

### LINKED ISSUE #59: Fixing pytests for email_notifier
# Description
Two tests are still currently failing and needs to be debugged!

# Outcome
- test_send_email_failure - AssertionError: expected call not found.
- test_check_and_notify_price_drop - AssertionError: Expected 'send_email' to be called once. Called 0 times.

# User Story
As an Engineer, when I run the `email_notifier.py` file, I want to be able to send notifications to the subscribed user and ensure functionality of that feature works smoothly!

### PR DESCRIPTION
**Pull request by indiah444**
- Fixed tests in `test_email_notifier.py`, all tests passing.

This closes #59

### CODE REVIEW

File: pipeline/test_email_notifier.py [hunk 1]
```diff
@@ -144,7 +141,7 @@ def test_check_and_notify_no_price_drop():
     mock_cursor.fetchall.return_value = [
         (1, 10.0, "Product A", "user@example.com")]
     mock_cursor.fetchone.return_value = {
-        "original_price": 15.0}
+        "price": 15.0}
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:144]: Nice fix from the other file too!

File: pipeline/test_email_notifier.py [hunk 2]
```diff
@@ -94,33 +93,33 @@ def test_send_email_failure():
...
-        "ResponseMetadata": {"HTTPStatusCode": 200}}
 
     with mock.patch.dict("os.environ", {
         "FROM_EMAIL": "test@example.com",
         "AWS_ACCESS_KEY_ID": "fake_key_id",
         "AWS_SECRET_ACCESS_KEY": "fake_secret_key"
     }), mock.patch("email_notifier.get_connection", return_value=mock_connection), \
-            mock.patch("email_notifier.get_ses_client", return_value=mock_ses_client), \
-            mock.patch("email_notifier.logging.info") as mock_info:
+            mock.patch("email_notifier.get_ses_client", return_value=mock_ses_client):
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:122]: Nice fix!

File: pipeline/test_email_notifier.py [hunk 3]
```diff
@@ -94,33 +93,33 @@ def test_send_email_failure():
     }), mock.patch("email_notifier.get_ses_client", return_value=mock_ses_client), \
             mock.patch("email_notifier.logging.error") as mock_error:
         send_email("user@example.com", "Subject", "Body")
+        sleep(0.1)
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:96]: Now we know why the error happened!

File: pipeline/test_email_notifier.py [hunk 4]
```diff
@@ -82,9 +83,7 @@ def test_send_email_success():
 
 
 def test_send_email_failure():
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:85]: Can you add a docstring:
""""Test that an error is logged when the email fails to send."""

File: pipeline/test_email_notifier.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 """Tests for email_notifier.py"""
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:2]: Can you add 
# pylint: skip-file
```suggestion
# pylint: skip-file
```

**Review by gem09lo — COMMENTED** [review]
Finally fixed the tests! Just some small changes left!

File: pipeline/test_email_notifier.py (hunk 3 above)
Comment (indiah444) [reply pipeline/test_email_notifier.py:96]: Finally!

**Review by gem09lo — APPROVED** [review]
Good changes thanks!

**Pull request merged**